### ISSUE TITLE
Genre mapping for library & archives collections

### ISSUE THREAD

**Issue by flapka** [ycba-cia/blacklight-collections2#281]
This is a variation of issue #207 . 

I’m returning my attention this week to how we might best map library and archives data to our Blacklight genre field. Part of this exercise could involve adapting some library terms to the concise set of genre terms used in the art collections (abstract art, animal art, architectural subject …). Since that list is admirably finite (18 terms), it wouldn’t be much trouble to take the variant terms used in MARC and map them to the TMS terms. In some cases the relationship will be one-to-one. In other cases the term in MARC will be a narrower concept.

For example:
•	botanical illustrations (MARC) --> botanical subject (TMS)
•	portraits (MARC) --> portrait (TMS)
•	maps (MARC) --> cartographic material (TMS)
•	atlases (MARC) --> cartographic material (TMS)

Other terms used in library / archive records won’t have art equivalents, in which case they’d map to genre as-is. For example:
•	diaries
•	artists’ books
•	auction catalogs

In terms of the XSLT, the approach could be as follows, I think:
•	For each term in a genre field, first evaluate whether there is a matching term from the TMS vocabulary (via 2 or 3 dozen when statements, perhaps). Apply the TMS string if there is a match
•	If there is no such match, then map the term as is, if the associated thesaurus code says that the term is within scope of our genre facet.

Conceptually and pragmatically, does this sound desirable and doable? I think just about everything described above could apply to genre data from ArchivesSpace as well.

**Comment by edgartdata** [comment]
@flapka It seems to me that the other terms (at least the 3 you mention above) might map well to a combination of classification and work types for the art collection? Here are the work types we currently use for the art collection (the list keeps growing): (I will add the list of classifications soon)

- painting
- drawing
- watercolor
- collage
- miniature
- intaglio print
- relief print
- planographic print
- photomechanical print
- photograph
- digital print
- altarpiece – a sculpture intended for liturgical use above an altar
- bozzetto/ sketch – a preliminary sketch for a finished work
- bust – a head sculpted in the round which may also include shoulders and arms
- herm – a bust that terminates and rests on a squared shape rather than a separate support
- high relief – a relief where more than half the mass of the sculpture projects into space.
- low relief – a relief where less than half the mass of the sculpture projects into space. 
- medallion – a largely flat, circular sculpture in relief 
- misericord – the projecting underside of a seat in the choir of a church 
- model/ modello/ maquette –a preparatory or presentation model more finished than a sketch
- overmantel – an ornamental sculpture intended to be displayed over a mantelpiece
- sculpture – all 3D objects in the Sculpture department must also have sculpture as an object name in addition to a more granular one, such as statue or bust.
- statue – a full sized, three-dimensional sculpture typically of the human figure
- statuette – a small sized, three-dimensional sculpture typically of the human figure but also encompassing animals  

And here is the list of our current genres, with scope notes: 
[genre termsYCBA.docx](https://github.com/ycba-cia/blacklight-collections2/files/5569527/genre.termsYCBA.docx)

**Comment by flapka** [comment]
@edgartdata I think this would benefit from discussion in one of our upcoming meetings. The library and archive descriptions include terms that categorize the *content* of the material (rather than the form), so with this in mind it very much would make sense to map such terms to genre rather than work type. In the three examples I listed previously, "artists books" was not the best, as it probably is better considered a work type. But auction catalogs and diaries are very much categorizations of content. To these, I'll add further common examples, such as:
* correspondence
* advertisements
* games
* poems
* caricatures

At the moment, I'm posing this question to identify the preferred long-term outcomes. Before implementation, we may need to do a fair bit of data remediation.

**Comment by edgartdata** [comment]
@flapka Uploading [YCBA TMS GENRES and their identifiers_July 2021.xlsx](https://github.com/ycba-cia/blacklight-collections2/files/6818795/YCBA.TMS.GENRES.and.their.identifiers_July.2021.xlsx) so that the RBMS and archives collections can use the same genre terms and the AAT IDs for them as the art collection. 

Please note that I will circle back to the AAT to double check if they have added new IDs that may be helpful for our genres: https://www.getty.edu/research/tools/vocabularies/aat/

**Comment by flapka** [comment]
Thanks @edgartdata , this is helpful. Do all of the terms have YCBA identifiers, including those that also have AAT identifiers? Or do you have YCBA identifiers only if the there's no equivalent concept in AAT?

**Comment by yulgit1** [comment]
@flapka The latter (unless something is getting pulled from another table which is unlikely). 

I like the mapping idea you proposed earlier (Nov 17 in this issue).  But there are > 100 terms, not 2 or 3 dozen making it a heavy task:
[marc_genreform.txt](https://github.com/ycba-cia/blacklight-collections2/files/6824073/marc_genreform.txt)

But along your earlier lines, part of that mapping could be sorting form_genre into object_name_ss (Work Type) or genre_name_ss (Genre) depending on what makes sense for the term.  In blacklight these fields are facets, but not displayed in item data.  For library objects in blacklight, that form_genre field isn't a facet but is displayed in item data.  If we go ahead with this it also might be good to consider harmonizing what's displayed into item data with perhaps both library and museum displaying the WorkType/Genre break out on the item page and facets.

**Comment by flapka** [comment]
I think I largely agree with @yulgit1 ’s assessment immediately above. 

@yulgit1 : Let’s imagine that there are 300 or so form/genre categories (MARC 655) used in YCBA records. And let’s imagine that I review that list of 300 to sort those that describe genre versus those that describe work type. If we had a table that recorded such sorting, is it conceivable that the table could be consulted for mapping from MARC to Blacklight? Or would that be too difficult / slow indexing down too much?

**Comment by yulgit1** [comment]
@flapka That should work, no performance issues.  Rather than a DB though, I would use a simple properties file with all the mappings in a format like:

botanical illustrations=botanical subject
portraits=portrait
maps=cartographic material
atlases=cartographic material

This would be a format java could readily read.

**Comment by yulgit1** [comment]
...also for each file this would include equivalences rather that default so:

diaries=diaries
auction catalogs=auction catalogs

so for the 2 properties files, work type and genre, there would only be 1 iteration through all the MARC 655s, that sorts by the mappings, and would ignore if not mapped.

**Comment by flapka** [comment]
@yulgit1 (This sounds good, but I concede the possibility that I don't fully follow your suggestion.) The ingest/mapping of MARC field 655 (genre/form) to Blacklight could perhaps accomplish three things:

1. Sort a term to genre or work type, based on our pre-determined list(s)
2. Where a MARC term has an analogous TMS genre concept, apply the TMS term (e.g. botanical illustrations = botanical subject)
3. Where a MARC term does not have analogous TMS concept, provide a specification for how the term should be converted from the native plural form to singular (e.g. games = game, prospectuses = prospectus)

Is that right / doable?

**Comment by edgartdata** [comment]
@flapka @yulgit1 There are 49 work types that we use in TMS. Would you like the list for the mapping?

**Comment by yulgit1** [comment]
@flapka It's very close, let me try restating what I was thinking:

First, create 2 mapping files, one will map 655->work type, the one will map 655->genre. Each mapping file will be inclusive of every term to be mapped, and the mapping will be either to the 49 TMS vocabulary or itself (singularized where that applies).  Then just iterate through and apply both mapping files to each 655.  The mappings will exist exclusively in one file or the other, effectively mapping every term to either normalized work type or genre.

Does that make sense?

**Comment by flapka** [comment]
@yulgit1 I think that makes sufficient sense. 

@edgartdata Yes, the list of 49 work types would be handy. Thanks!

**Comment by edgartdata** [comment]
@flapka Here is the list of TMS art collection object names/work types active as of August 6, 2021 (definitions are in the TMS cataloguing manual link in the CIA SharePoint page):

- Altarpiece
- Bench
- Bookplate
- Bozzetto/sketch
- Bust
- Buttons
- Cake
- Ceramic
- Cigar box
- Collage
- Digital print
- Drawing
- Frame
- Frontispiece
- Herm
- High relief
- Installation
- Intaglio print
- lantern slide 
- Letter box
- Letterpress
- Low relief
- Maquette
- Medallionminiature
- Misericord
- Model
- Overmantel
- Painting
- Palette
- Photograph
- Photomechanical print
- Planographic print
- Poster
- Relief print
- Rubbing
- Screen (furniture)
- Sculpture
- Sketchbook
- Snuff box
- Statue
- Statuette
- Tableau
- Ticket
- Trade card
- Watercolor
- Waxwork (sculpture)

**Comment by flapka** [comment]
Thanks @edgartdata. Yoonha and I will work on the mapping in the coming weeks. Hope to have something to share in September.

**Comment by edgartdata** [comment]
@flapka I guess i will need to give you the AAT IDs we are using for them, and our local YCBA IDs when there are no good AAT semantic matches. Would you be using these IDs too? Is that the idea?

**Comment by flapka** [comment]
@edgartdata Yes, the AAT IDs could be useful, if you could share them. That said, we'd want the mapping to function even where those IDs are absent (i.e. matching on labels alone), since many legacy records will lack IDs -- and revision of all existing records would be a difficult project, given Voyager shortcomings.

Does our Blacklight index include field(s) for such IDs? If so, adding them after mapping would appear to make sense. But I don't see a work type (object name) ID field in this summary: https://docs.google.com/spreadsheets/d/1aZ6s3re_kT5NlFmc2UPEsbjq_l0x5b0BMydGre36vOg/edit#gid=0

**Comment by yulgit1** [comment]
The blacklight index does not contain authority IDs.  But in the spreadsheet work type is row 19 object_name_ss,
